### PR TITLE
Remove Components from list that are not for images

### PR DESCRIPTION
**Pull request by tmjd**
## Description

This PR drops the "Components" from the EnterpriseComponents (EnterpriseImages) that are only used for specifying a version of a component (like for elasticsearch or prometheus) that needs to be passed to an operator for configuration of those components version. (for example, we have to pass a version of ES to the eck-operator)

Output of `operator --print-images=list` is now
```
docker.io/calico/cni:master
docker.io/calico/kube-controllers:master
docker.io/calico/node:master
docker.io/calico/typha:master
docker.io/calico/pod2daemon-flexvol:master
quay.io/tigera/operator:v1.23.0-1.dev-447-gce924bb721de-dirty
docker.io/calico/apiserver:master
docker.io/calico/windows-upgrade:master
gcr.io/unique-caldron-775/cnx/tigera/cnx-apiserver:master
gcr.io/unique-caldron-775/cnx/tigera/compliance-benchmarker:master
gcr.io/unique-caldron-775/cnx/tigera/compliance-controller:master
gcr.io/unique-caldron-775/cnx/tigera/compliance-reporter:master
gcr.io/unique-caldron-775/cnx/tigera/compliance-server:master
gcr.io/unique-caldron-775/cnx/tigera/compliance-snapshotter:master
gcr.io/unique-caldron-775/cnx/tigera/deep-packet-inspection:master
gcr.io/unique-caldron-775/cnx/tigera/intrusion-detection-job-installer:master
gcr.io/unique-caldron-775/cnx/tigera/elasticsearch:master
gcr.io/unique-caldron-775/cnx/tigera/eck-operator:master
gcr.io/unique-caldron-775/cnx/tigera/es-curator:master
gcr.io/unique-caldron-775/cnx/tigera/es-proxy:master
gcr.io/unique-caldron-775/cnx/tigera/fluentd:master
gcr.io/unique-caldron-775/cnx/tigera/fluentd-windows:master
gcr.io/unique-caldron-775/cnx/tigera/guardian:master
gcr.io/unique-caldron-775/cnx/tigera/intrusion-detection-controller:master
gcr.io/unique-caldron-775/cnx/tigera/anomaly_detection_jobs:master
gcr.io/unique-caldron-775/cnx/tigera/anomaly-detection-api:master
gcr.io/unique-caldron-775/cnx/tigera/kibana:master
gcr.io/unique-caldron-775/cnx/tigera/cnx-manager:master
gcr.io/unique-caldron-775/cnx/tigera/dex:master
gcr.io/unique-caldron-775/cnx/tigera/voltron:master
gcr.io/unique-caldron-775/cnx/tigera/packetcapture:master
gcr.io/unique-caldron-775/cnx/tigera/l7-collector:master
gcr.io/unique-caldron-775/cnx/tigera/envoy:master
gcr.io/unique-caldron-775/cnx/tigera/prometheus:master
gcr.io/unique-caldron-775/cnx/tigera/prometheus-service:master
gcr.io/unique-caldron-775/cnx/tigera/alertmanager:master
gcr.io/unique-caldron-775/cnx/tigera/cnx-queryserver:master
gcr.io/unique-caldron-775/cnx/tigera/kube-controllers:master
gcr.io/unique-caldron-775/cnx/tigera/cnx-node:master
gcr.io/unique-caldron-775/cnx/tigera/typha:master
gcr.io/unique-caldron-775/cnx/tigera/cni:master
gcr.io/unique-caldron-775/cnx/tigera/cloud-controllers:master
gcr.io/unique-caldron-775/cnx/tigera/elasticsearch-metrics:master
gcr.io/unique-caldron-775/cnx/tigera/es-gateway:master
gcr.io/unique-caldron-775/cnx/tigera/calico-windows-upgrade:master
gcr.io/unique-caldron-775/cnx/tigera/dikastes:master
quay.io/tigera/key-cert-provisioner:master
```
Before this PR the output is
```
docker.io/calico/cni:master
docker.io/calico/kube-controllers:master
docker.io/calico/node:master
docker.io/calico/typha:master
docker.io/calico/pod2daemon-flexvol:master
quay.io/tigera/operator:v1.23.0-1.dev-446-g2c324e7d5e35
docker.io/calico/apiserver:master
docker.io/calico/windows-upgrade:master
gcr.io/unique-caldron-775/cnx/tigera/cnx-apiserver:master
gcr.io/unique-caldron-775/cnx/tigera/compliance-benchmarker:master
gcr.io/unique-caldron-775/cnx/tigera/compliance-controller:master
gcr.io/unique-caldron-775/cnx/tigera/compliance-reporter:master
gcr.io/unique-caldron-775/cnx/tigera/compliance-server:master
gcr.io/unique-caldron-775/cnx/tigera/compliance-snapshotter:master
gcr.io/unique-caldron-775/cnx/tigera/deep-packet-inspection:master
gcr.io/unique-caldron-775/cnx/tigera/elasticsearch:7.16.2
gcr.io/unique-caldron-775/cnx/tigera/kibana:7.16.2
gcr.io/unique-caldron-775/cnx/tigera/intrusion-detection-job-installer:master
gcr.io/unique-caldron-775/cnx/tigera/elasticsearch:master
gcr.io/unique-caldron-775/cnx/tigera/eck-operator:1.8.0
gcr.io/unique-caldron-775/cnx/tigera/eck-operator:master
gcr.io/unique-caldron-775/cnx/tigera/es-curator:master
gcr.io/unique-caldron-775/cnx/tigera/es-proxy:master
gcr.io/unique-caldron-775/cnx/tigera/fluentd:master
gcr.io/unique-caldron-775/cnx/tigera/fluentd-windows:master
gcr.io/unique-caldron-775/cnx/tigera/guardian:master
gcr.io/unique-caldron-775/cnx/tigera/intrusion-detection-controller:master
gcr.io/unique-caldron-775/cnx/tigera/anomaly_detection_jobs:master
gcr.io/unique-caldron-775/cnx/tigera/anomaly-detection-api:master
gcr.io/unique-caldron-775/cnx/tigera/kibana:master
gcr.io/unique-caldron-775/cnx/tigera/cnx-manager:master
gcr.io/unique-caldron-775/cnx/tigera/dex:master
gcr.io/unique-caldron-775/cnx/tigera/voltron:master
gcr.io/unique-caldron-775/cnx/tigera/packetcapture:master
gcr.io/unique-caldron-775/cnx/tigera/l7-collector:master
gcr.io/unique-caldron-775/cnx/tigera/envoy:master
gcr.io/unique-caldron-775/cnx/tigera/prometheus:v2.32.0
gcr.io/unique-caldron-775/cnx/tigera/prometheus:master
gcr.io/unique-caldron-775/cnx/tigera/prometheus-service:master
gcr.io/unique-caldron-775/cnx/tigera/alertmanager:v0.23.0
gcr.io/unique-caldron-775/cnx/tigera/alertmanager:master
gcr.io/unique-caldron-775/cnx/tigera/cnx-queryserver:master
gcr.io/unique-caldron-775/cnx/tigera/kube-controllers:master
gcr.io/unique-caldron-775/cnx/tigera/cnx-node:master
gcr.io/unique-caldron-775/cnx/tigera/typha:master
gcr.io/unique-caldron-775/cnx/tigera/cni:master
gcr.io/unique-caldron-775/cnx/tigera/cloud-controllers:master
gcr.io/unique-caldron-775/cnx/tigera/elasticsearch-metrics:master
gcr.io/unique-caldron-775/cnx/tigera/es-gateway:master
gcr.io/unique-caldron-775/cnx/tigera/calico-windows-upgrade:master
gcr.io/unique-caldron-775/cnx/tigera/dikastes:master
quay.io/tigera/key-cert-provisioner:master
```

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/components/enterprise.go [hunk 1]
```diff
@@ -57,12 +57,10 @@ var (
 
 	ComponentEckElasticsearch = component{
 		Version: "7.16.2",
-		Image:   "tigera/elasticsearch",
```
Comment (Brian-McM) [inline pkg/components/enterprise.go:60]: Why is this removed?

File: hack/gen-versions/components.go [hunk 2]
```diff
@@ -32,11 +32,11 @@ var defaultImages = map[string]string{
 	"flannel":                    "coreos/flannel",
 	"flexvol":                    "calico/pod2daemon-flexvol",
 	"typha":                      "calico/typha",
-	"eck-elasticsearch":          "tigera/elasticsearch",
-	"eck-elasticsearch-operator": "tigera/eck-operator",
-	"eck-kibana":                 "tigera/kibana",
-	"coreos-prometheus":          "tigera/prometheus",
-	"coreos-alertmanager":        "tigera/alertmanager",
+	"eck-elasticsearch":          "unused/image",
```
Comment (Brian-McM) [inline hack/gen-versions/components.go:35]: Why are we changing this to unused? We still install these images, defaults are still nice to have...

Comment (tmjd) [reply hack/gen-versions/components.go:35]: We don't use these images. Take a look at the entries in the versions yaml https://github.com/tigera/operator/blob/master/config/enterprise_versions.yml, we don't include an image for the entries changed to `unused/image` because we only use the version field for those entries. You'll find a similar entry without `eck` or `coreos` that we do use that has an image field.

File: pkg/components/enterprise.go (hunk 1 above)
Comment (tmjd) [reply pkg/components/enterprise.go:60]: This was removed because this ComponentEckElasticsearch is only used to specify the version of Elasticsearch (in this case 7.16.2) so we can provide that to the ES operator. The ComponentElasticsearch has the image we use which can be found at line 73.

File: hack/gen-versions/components.go (hunk 2 above)
Comment (Brian-McM) [reply hack/gen-versions/components.go:35]: Right right, thanks for the explanation.

**Review by Brian-McM — APPROVED** [review]
